engine/src/main/java/org/hibernate/validator/internal/util/DomainNameUtil.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.util;
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+
+import java.net.IDN;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Marko Bekhta
+ */
+public final class DomainNameUtil {
+
+	/**
+	 * This is the maximum length of a domain name. But be aware that each label (parts separated by a dot) of the
+	 * domain name must be at most 63 characters long. This is verified by {@link IDN#toASCII(String)}.
+	 */
+	private static final int MAX_DOMAIN_PART_LENGTH = 255;
+
+	private static final String DOMAIN_CHARS_WITHOUT_DASH = "[a-z\u0080-\uFFFF0-9!#$%&'*+/=?^_`{|}~]";
+	private static final String DOMAIN_LABEL = "(" + DOMAIN_CHARS_WITHOUT_DASH + "-?)*" + DOMAIN_CHARS_WITHOUT_DASH + "+";
+
+	private static final String DOMAIN = DOMAIN_LABEL + "+(\\." + DOMAIN_LABEL + "+)*";
+
+	private static final String IP_DOMAIN = "[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}";
+	//IP v6 regex taken from http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses
+	private static final String IP_V6_DOMAIN = "(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))";
+
+	/**
+	 * Regular expression for the domain part of an URL
+	 */
+	// A host string must be a domain string, an IPv4 address string, or "[", followed by an IPv6 address string, followed by "]".
+	private static final Pattern DOMAIN_PATTERN = Pattern.compile(
+			DOMAIN + "|\\[" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
+	);
+
+	/**
+	 * Regular expression for the domain part of an email address (everything after '@')
+	 */
+	private static final Pattern EMAIL_DOMAIN_PATTERN = Pattern.compile(
+			DOMAIN + "|\\[" + IP_DOMAIN + "\\]|" + "\\[IPv6:" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
+	);
+
+	private DomainNameUtil() {
+	}
+
+	/**
+	 * Checks validity of domain name used in email. To be valid it should be either valid host name, or IP address in []
+	 *
+	 * @param domain domain to check for validity
+	 *
+	 * @return {@code true} if provided string is a valid domain, {@code false} otherwise
```

Comment:
In such moments I sometimes think - maybe I should use Eclipse (at least for HV), to not have these formatting differences... But then I understand that I'm to used to IntelliJ :)
need to find and update this to match :)